jobs/basic-company/main.go

```diff
@@ -0,0 +1,178 @@
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"math/rand"
+	"time"
+
+	"golang.org/x/oauth2/google"
+	jobs "google.golang.org/api/jobs/v2"
+)
+
+const requestDeadline = 10 * time.Second
+
+func prettyFormat(v interface{}) string {
+	j, err := json.MarshalIndent(v, "", "    ")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return string(j)
+}
+
+// [START basic_company]
+
+func makeCompany() *jobs.Company {
+	return &jobs.Company{
+		DisplayName:          "Google",
+		HqLocation:           "1600 Amphitheatre Parkway Mountain View, CA 94043",
+		DistributorCompanyId: fmt.Sprintf("company:%d", rand.New(rand.NewSource(time.Now().UnixNano())).Int63()),
+	}
+}
+
+// [END basic_company]
+
+// [START create_company]
+
+func createCompany(ctx context.Context, js *jobs.Service, toCreate *jobs.Company) (*jobs.Company, error) {
+	fmt.Println("Attempting to create a Company...")
+
+	createCtx, cancel := context.WithTimeout(ctx, requestDeadline)
+	defer cancel()
+
+	c, err := js.Companies.Create(toCreate).Context(createCtx).Do()
+	if err != nil {
+		return nil, err
+	}
+
+	fmt.Printf("Company created:\n %v\n", prettyFormat(c))
+	return c, nil
+}
+
+// [END create_company]
+
+// [START get_company]
+
+func getCompany(ctx context.Context, js *jobs.Service, name string) (*jobs.Company, error) {
+	fmt.Printf("Attempting to get a Company with name %s...\n", name)
+
+	getCtx, cancel := context.WithTimeout(ctx, requestDeadline)
+	defer cancel()
+
+	c, err := js.Companies.Get(name).Context(getCtx).Do()
+	if err != nil {
+		return nil, err
+	}
+
+	fmt.Printf("Company retrieved:\n %v\n", prettyFormat(c))
+	return c, nil
+}
+
+// [END get_company]
+
+// [START update_company]
+
+func patchCompany(ctx context.Context, js *jobs.Service, company *jobs.Company) (*jobs.Company, error) {
+	return patchCompanyWithFieldMask(ctx, js, company, "")
+}
+
+// [END update_company]
+
+// [START update_company_with_field_mask]
+
+func patchCompanyWithFieldMask(ctx context.Context, js *jobs.Service, company *jobs.Company, fields string) (*jobs.Company, error) {
+	fmt.Printf("Attempting to patch a Company with name %s...\n", company.Name)
+
+	patchCtx, cancel := context.WithTimeout(ctx, requestDeadline)
+	defer cancel()
+
+	req := js.Companies.Patch(company.Name, company).Context(patchCtx)
+	if fields != "" {
+		req.UpdateCompanyFields(fields)
+	}
+
+	c, err := req.Do()
+	if err != nil {
+		return nil, err
+	}
+
+	fmt.Printf("Company updated:\n %v\n", prettyFormat(c))
+	return c, nil
+}
+
+// [END update_company_with_field_mask]
+
+// [START delete_company]
+func deleteCompany(ctx context.Context, js *jobs.Service, name string) error {
+	fmt.Printf("Attempting to delete a Company with name %s...\n", name)
+
+	deleteCtx, cancel := context.WithTimeout(ctx, requestDeadline)
+	defer cancel()
+
+	if _, err := js.Companies.Delete(name).Context(deleteCtx).Do(); err != nil {
+		return err
+	}
+
+	fmt.Println("Company deleted")
+	return nil
+}
+
+// [END delete_company]
+
+func main() {
```

Comment:
We have some new guidance to not have a `package main` for snippets (https://github.com/GoogleCloudPlatform/golang-samples/issues/530). Would you mind removing this? Each snippet would need to get the jobs service itself and can `log.Fatal`.